drivers/adc/ad7124/ad7124.h

```diff
@@ -257,19 +265,139 @@
 #define AD7124_FILT_REG_SINGLE_CYCLE      (1 << 16)
 #define AD7124_FILT_REG_FS(x)             (((x) & 0x7FF) << 0)
 
+#define AD7124_CRC8_POLYNOMIAL_REPRESENTATION 0x07 /* x8 + x2 + x + 1 */
+#define AD7124_DISABLE_CRC 0
+#define AD7124_USE_CRC 1
+#define AD7124_CHMAP_REG_SETUP_SEL_MSK  	NO_OS_GENMASK(14,12)
+#define AD7124_CHMAP_REG_AINPOS_MSK    		NO_OS_GENMASK(9,5)
+#define AD7124_CHMAP_REG_AINNEG_MSK    		NO_OS_GENMASK(4,0)
+#define AD7124_ADC_CTRL_REG_MODE_MSK   		NO_OS_GENMASK(5,2)
+#define AD7124_SETUP_CONF_REG_REF_SEL_MSK	NO_OS_GENMASK(4,3)
+#define AD7124_REF_BUF_MSK                  NO_OS_GENMASK(8,7)
+#define AD7124_AIN_BUF_MSK                  NO_OS_GENMASK(6,5)
+
 /******************************************************************************/
 /*************************** Types Declarations *******************************/
 /******************************************************************************/
+/**
+ * @enum	ad7124_device_type
+ * @details AD7124 Device definitions
+**/
+enum ad7124_device_type {
+	ID_AD7124_4,
+	ID_AD7124_8
+};
+
+/**
+ * @enum  ad7124_mode
+ * @brief ADC Modes of Operation
+**/
+enum ad7124_mode {
+	AD7124_CONTINUOUS,
+	AD7124_SINGLE,
+	AD7124_STANDBY,
+	AD7124_POWER_DOWN,
+	AD7124_IDLE,
+	AD7124_IN_ZERO_SCALE_OFF,
+	AD7124_IN_FULL_SCALE_GAIN,
+	AD7124_SYS_ZERO_SCALE_OFF,
+	AD7124_SYS_ZERO_SCALE_GAIN,
+	ADC_MAX_MODES
+};
+
+/**
+ * @enum ad7124_analog_input
+ * @brief ADC input sources for each channel.
+ **/
+enum ad7124_analog_input {
```

Comment:
No need to explicitly declare the value of each enum. Only declare explicit value for enum member that would be breaking the sequence i.e., if next enum value from 0x06 would be 0x10.

Reply:
Will change that.